lib/source_availability_checker.rb

```diff
@@ -0,0 +1,80 @@
+class SourceAvailabilityChecker
+  attr_accessor :thread_available_checker, :thread_unavailable_checker
+
+  def self.instance
+    @instance ||= new
+  end
+
+  def initialize
+    start_threads
+  end
+  private_class_method :new
+
+  private
+
+  def log
+    Rails.logger
+  end
+
+  def start_threads
+    @thread_available_checker = Thread.new do
+      interval = ENV["AVAILABLE_SOURCE_CHECK_INTERVAL"] || 60
+      loop do
+        sleep(interval)
+        check_available_sources
+      end
+    end
+
+    @thread_unavailable_checker = Thread.new do
+      interval = ENV["UNAVAILABLE_SOURCE_CHECK_INTERVAL"] || 10
+      loop do
+        sleep(interval)
+        check_unavailable_sources
+      end
+    end
+  end
+
+  def check_available_sources
+    check_sources = []
+
+    Source.includes(:source_type, :tenant).all.each do |source|
+      next unless source_available(source)
+
+      check_sources << source.id
+      Sources::Api::Events.send_event(
```

Comment:
This isn't really an event at this point since it is going to the operations worker it is more of a queue message, I know it is just naming but the Event class was supposed to be very specific not just reusing the kafka connection.